### ISSUE TITLE
[Build stability] TestInfo

### ISSUE THREAD

**Issue by nikkiki** [sensu/sensu-go#1829]
`host.Info()` can return an error in Appveyor due to `context deadline exceeded`. The context is provided by https://github.com/shirou/gopsutil/blob/master/host/host_windows.go#L43-L45
```
--- FAIL: TestInfo (3.00s)
	assertions.go:239: 
                          
	Error Trace:	system_test.go:13
		
	Error:      	Received unexpected error:
		
	            	context deadline exceeded
	assertions.go:239: 
                          
	Error Trace:	system_test.go:14
		
	Error:      	Should NOT be empty, but was 
	assertions.go:239: 
                          
	Error Trace:	system_test.go:15
		
	Error:      	Should NOT be empty, but was 
	assertions.go:239: 
                          
	Error Trace:	system_test.go:16
		
	Error:      	Should NOT be empty, but was 
	assertions.go:239: 
                          
	Error Trace:	system_test.go:17
		
	Error:      	Should NOT be empty, but was 
	assertions.go:239: 
                          
	Error Trace:	system_test.go:21
		
	Error:      	Should NOT be empty, but was 
	assertions.go:239: 
                          
	Error Trace:	system_test.go:22
		
	Error:      	Should NOT be empty, but was []
```

**Comment by amdprophet** [comment]
At a glance it looks like this may be due to gopsutil hanging on `windows/386`. There have been a few updates to the windows code between `2.18.04` and `2.18.06`:
https://github.com/shirou/gopsutil/compare/v2.18.04...v2.18.06. It may be worth trying the latest version to see if it resolves the issue.

**Comment by amdprophet** [comment]
This is happening due to WMI queries taking too long https://github.com/shirou/gopsutil/blob/v2.18.04/internal/common/common_windows.go#L119. The timeout is 3 seconds which feels long already. I'm not sure we'll want to increase the timeout as it will have a direct effect on how long it takes to send a keepalive.

We'll need to figure out how to guard against cases where we hit a timeout for WMI queries. We may want to support keepalive messages that may be missing system information - so long as we make sure not to override any previously stored system information with nothingness.

One other thing we may want to do is collect system information out-of-band of the `getAgentEntity()` function in the agent so that we aren't holding up keepalive messages when trying to fetch system info.

Any thoughts?

**Comment by grepory** [comment]
@amdprophet Can you open a ticket for the `getAgentEntity()` enhancement? I think that's a good call. That allows us to timeout while still handling keepalives. I think it's okay if system info is stale for a bit.

**Comment by amdprophet** [comment]
I've created https://github.com/sensu/sensu-go/issues/1947 to track the issue.

**Comment by amdprophet** [comment]
This should be fixed by #1947.
